src/FluentValidation.Tests/UserSeverityTester.cs

```diff
@@ -44,5 +43,50 @@ public UserSeverityTester()
 		    var result = validator.Validate( new Person() );
 		    result.Errors.Single().Severity.ShouldEqual( Severity.Error );
 		}
+
+		[Fact]
+		public void Throws_when_provider_is_null() {
+			typeof(ArgumentNullException).ShouldBeThrownBy(() => validator.RuleFor(x => x.Surname).NotNull().WithSeverity((Func<Person, Severity>) null));
+		}
+
+		[Fact]
+		public void Correctly_provides_object_being_validated() {
+			Person resultPerson = null;
+
+			validator.RuleFor(x => x.Surname).NotNull().WithSeverity(x => {
+				resultPerson = x;
+				return Severity.Warning;
+			});
+
+			var person = new Person();
+			validator.Validate(person);
+
+			resultPerson.ShouldBeTheSameAs(person);
+		}
+
+		[Fact]
+		public void Can_Provide_severity_for_item_in_collection() {
+			validator.RuleForEach(x => x.Children).NotNull().WithSeverity((person, child) => Severity.Warning);
+			var result = validator.Validate(new Person {Children = new List<Person> {null}});
+			result.Errors[0].Severity.ShouldEqual(Severity.Warning);
+		}
+
+		[Fact]
+		public void Can_Provide_conditional_severity() {
+			validator.RuleFor(x => x.Surname).NotNull().WithSeverity(x => x.Age > 10 ? Severity.Info : Severity.Warning);
+
+			var person = new Person();
+
+			{

```

Comment:
Remove these braces please - can just redefine the `result` variable.

Reply:
I prefer to not reuse result variables in tests, but will change it.